packages/admin-app/src/pages/LoginPage.tsx

```diff
@@ -1,16 +1,117 @@
-import { Box } from '@material-ui/core';
-import React from 'react';
+import {
+    Body_login_for_access_token_api_v1_auth_login_post,
+    AuthClient,
+    UrlGetter,
+    ErrorMessage,
+} from '@electionguard/api-client';
+import { Button, Container, InputAdornment, makeStyles, TextField } from '@material-ui/core';
+import { AccountCircle, Lock } from '@material-ui/icons';
+import React, { useState } from 'react';
 
-export const LoginPage: React.FC = () => (
-    <Box
-        height="100%"
-        display="flex"
-        flexDirection="column"
-        justifyContent="center"
-        alignItems="center"
-    >
-        Login
-    </Box>
-);
+const useStyles = makeStyles((theme) => ({
+    root: {
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    errorMessage: {
+        color: 'red',
+        marginBottom: theme.spacing(3),
+    },
+    text: {
+        marginBottom: theme.spacing(3),
+        width: '100%',
+    },
+    submit: {
+        marginTop: theme.spacing(3),
+        width: '100%',
+    },
+}));
+
+export interface LoginPageProps {
+    setToken: (token: string) => void;
+}
+
+export const LoginPage: React.FC<LoginPageProps> = ({ setToken }) => {
+    const classes = useStyles();
+    const [username, setUserName] = useState('');
+    const [password, setPassword] = useState('');
+    const [result, setResult] = useState<string>();
+
+    const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (e) => {
+        e.preventDefault();
+        const url = UrlGetter.GetUrl();
```

Comment:
Using the url here seems like poor encapsulation, I'll look into that

Reply:
Fixed.  Put in a ClientFactory that's responsible for instantiating the generated proxy classes and passing in the correct url.